tests/diff/test_diff_spec.lua

```diff
@@ -0,0 +1,156 @@
+-- Luagit2's Diff Module Tests
+
+local fixer = require("Fixtures.fix_repo")
+
+describe(" Diff Methods Tests ", function()
+    local luagit2 = require("luagit2")
+    local repo_path = "Fixtures/WORKON_REPO"
+    local diff_options, repo_index
+
+    setup(function()
+        luagit2.init()
+    end)
+    
+    before_each(function()
+        fixer.set_repo("new_test_repo")
+        repo = luagit2.repository_open(repo_path)
```

Comment:
`repo` is global here :) you can fix it in this PR, but I suggest that once all PRs are merged, that you run luacheck on the whole project just to see if there is anything else we didnt catch :)